app/assets/javascripts/discourse/app/components/topic-list-item.js

```diff
@@ -58,6 +58,16 @@ export default Component.extend({
         if (this.selected && this.selected.includes(this.topic)) {
           this.element.querySelector("input.bulk-select").checked = true;
         }
+        const mainLink = this.element.querySelector(".main-link");
+        const title = mainLink?.querySelector(".title");
+        if (mainLink && title) {
+          title.addEventListener("focus", () => {
```

Comment:
where is this removed?

Reply:
Do we still need to remove the listener if the element will be removed when the user navigates to a different page?

Reply:
it's most likely good, however, Im unsure what happens here. Your anonymous function holds reference to a DOM node, so Im not sure all of this will clean up nicely.

Reply:
Ok I've added cleanup for the listeners in the `willDestroyElement` hook. Better safe than sorry!